dataset/instrution_commit_clean.py

```diff
@@ -0,0 +1,312 @@
+
+    for i in range(1, 250):
+        downloaded_files.append("data/java/java-{:04d}.jsonl".format(i))
+    return downloaded_files
+
+
+data_files = prepare_download_files()
+
+
+def download_file(file):
+    global counter
+    print("start")
+    file = hf_hub_download(DATASET_NAME, file, repo_type="dataset",
+                           cache_dir=CACHE_DIR)
+    with counter.get_lock():
+        counter.value += 1
+    print(counter.value)
+    return file
+
+
+# download files using multi-thread
+with Pool(16, initializer=init, initargs=(counter,)) as p:
+    _ = p.map(download_file, data_files[::2])
+
+# obtain the file path
+files = [hf_hub_download(DATASET_NAME, file, repo_type="dataset",
+                         cache_dir=CACHE_DIR) for file in data_files[::2]]
+
+counter = Value('i', 0)
+
+
+def load_file(file):
+    global counter
+    print("start")
+    file = load_dataset("/".join(file.split("/")[:-1]), data_files=file,
+                        split="train", cache_dir=CACHE_DIR)
+    with counter.get_lock():
+        counter.value += 1
+    print(counter.value)
+    return file
+
+
+with Pool(8, initializer=init, initargs=(counter,)) as p:
+    ds_list = p.map(load_file, files)
+
+ds = concatenate_datasets(ds_list)
+
+print("The dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: x["proba"] >= 0.9, num_proc=30)
+
+print("After proba filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: len(x["old_contents"]) < 100_000, num_proc=30)
+
+print("After content length filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.map(get_line_diff_range, num_proc=30)
+
+
+def commit_filter(example):
+    if len(example["new_contents"]) == 0:
+        return False
+
+    if example["old_change_range"] <= 2:
+        if random.random() > SHORT_SAMPLING:
+            return False
+
+    if example["old_change_range"] >= 200:
+        if random.random() > LONG_SAMPLING:
+            return False
+
+    if example["old_file"].split(".")[-1] == "md":
+        if random.random() > MD_SAMPLING:
+            return False
+
+    if example["old_file"].split(".")[-1] in DATA_EXT:
+        if random.random() > DATA_SAMPLING:
+            return False
+
+    if len(example["subject"]) == 0 or len(example["subject"].split()) == 0:
+        return False
+
+    # remove samples with bad subwords
+    lower_subject = example["subject"].lower()
+    for bad_msg in BAD_SUB_MESSAGE:
+        if bad_msg in lower_subject:
+            return False
+
+    # remove samples with bad messages
+    for bad_msg in BAD_MESSAGE:
+        if bad_msg == lower_subject:
+            return False
+
+    # version updates (e.g. v1.1.0)
+    if re.match(r"(?:v)?\d+\.\d+\.\d+(?=$|\S)", lower_subject):
+        return False
+
+    # high character/token ratio (e.g. hashes)
+    if len(example["subject"]) / len(example["subject"].split()) > 20:
+        return False
```

Comment:
Nice idea, but I think this filters out most/all Chinese / Japanese commit messages - I think we could leave them in;
I will try to do some investigation into what exactly this filters out

Reply:
@Muennighoff This is used to filter out some strings like hashes, and this is also inspired by the fertility of tokenization in SantaCoder. BTW, I agree that most of Chinese / Japanese commit messages will be filtered out - maybe it should be during pre-training but can be saved in our fine-tuning. I will try another way to filter out hashes! Thanks again!